packages/eslint-plugin/tests/rules/unbound-method.test.ts

```diff
@@ -965,7 +993,7 @@ function foo({ unbound }: { unbound: () => string } | Foo) {}
       errors: [
         {
           line: 5,
-          messageId: 'unbound',
+          messageId: 'unboundWithoutThisAnnotation',
```

Comment:
I think this change is also correct here, the type the function expects is `{ unbound: () => string }` which is missing a `this: void` type annotation.

Reply:
Somewhat related: https://github.com/typescript-eslint/typescript-eslint/issues/11683.